pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -662,5 +663,6 @@ func InternalVersionDecoder() runtime.Decoder {
 }
 
 func InternalVersionJSONEncoder() runtime.Encoder {
-	return legacyscheme.Codecs.LegacyCodec(legacyscheme.Registry.EnabledVersions()...)
+	encoder := legacyscheme.Codecs.LegacyCodec(legacyscheme.Registry.EnabledVersions()...)
+	return unstructured.JSONFallbackEncoder{Encoder: encoder}
```

Comment:
This needs some doc.  Falling back to a JSON encoder could let us serialize an internal object to invalid json, right?